<?xml version="1.0"?>
<launch>

  <!-- Camera ID -->
  <arg name="device_id"               default="#1" />
  <arg name="camera_name"             default="kinect" />
  <arg name="rgb_camera_info_url"     default="" />
  <arg name="intermediate_topic"      default="/ground_based_people_detector_$(arg camera_name)/detections" />
  <arg name="ground_from_calibration" default="false" />
  
  <!-- Camera driver -->
  <include file="$(find detection)/launch/$(env KINECT_DRIVER).launch"> 
    <arg name="device_id"           value="$(arg device_id)" />
    <arg name="camera"              value="$(arg camera_name)" />
    <arg name="rgb_camera_info_url" value="$(arg rgb_camera_info_url)" />
  </include>

  <!-- Launch ground based people detection node -->
  <node pkg="detection" type="ground_based_people_detector"
        name="ground_based_people_detector_$(arg camera_name)" output="screen" respawn="true">
        
    <rosparam command="load"                        file="$(find detection)/conf/ground_based_people_detector.yaml" /> 
    
    <param name="classifier_file"                   value="$(find detection)/data/HogSvmPCL.yaml"/>
    <param name="pointcloud_topic"                  value="/$(arg camera_name)/depth_registered/points"/>
    <param name="output_topic"                      value="$(arg intermediate_topic)"/>
    <param name="camera_info_topic"                 value="/$(arg camera_name)/rgb/camera_info"/>
    <param name="rate"                              value="60.0"/>  
    <param name="ground_from_extrinsic_calibration" value="$(arg ground_from_calibration)"/>  
    
  </node>

  <!-- Launch HaarDispAda Node-->
  <node pkg="detection" type="HaarDispAda174" name="HaarDispAdaNode_$(arg camera_name)" output="screen" respawn="true">
  	
    <rosparam command="load"       file="$(find detection)/conf/haar_disp_ada_detector.yaml" /> 
    
    <remap from="Color_Image"      to="/$(arg camera_name)/rgb/image_color"/>	
    <remap from="Disparity_Image"  to="/$(arg camera_name)/depth_registered/disparity"/>
    <remap from="input_detections" to="$(arg intermediate_topic)"/>
    
    <param name="classifier_file"  value="$(find detection)/data/HaarDispAda.xml"/>
    
  </node>  

</launch>
